lib/reverse_sentence.rb

```diff
@@ -1,6 +1,15 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(1)
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  return nil if my_sentence == nil 
+
+  split_sentence = my_sentence.scan(/\S+|\s+/) 
```

Comment:
This creates an array which is based on the size of the string... So it's O(n) space complexity.